modules/seqexec/web/client/src/main/scala/seqexec/web/client/components/sequence/steps/NodAndShuffleProgress.scala

```diff
@@ -0,0 +1,266 @@
+import monocle.macros.Lenses
+import react.common.Css
+import seqexec.model.{NodAndShuffleStatus, ObservationProgress}
+import seqexec.model.operations._
+import seqexec.web.client.circuit.SeqexecCircuit
+import seqexec.web.client.model.{ClientStatus, StopOperation}
+import seqexec.web.client.reusability._
+import seqexec.web.client.semanticui._
+
+import scala.math.max
+
+final case class SmoothDividedProgressBar(
+  sections            : List[DividedProgress.Label],
+  sectionTotal        : DividedProgress.Quantity,
+  value               : DividedProgress.Quantity,
+  maxValue            : DividedProgress.Quantity,
+  completeSectionColor: Option[String] = None,
+  ongoingSectionColor : Option[String] = None,
+  progressCls         : List[Css] = Nil,
+  barCls              : List[Css],
+  labelCls            : List[Css] = Nil,
+  stopping            : Boolean,
+  paused: Boolean
+) extends SmoothProgressBarProps {
+  @inline def render: VdomElement = SmoothDividedProgressBar.component(this)
+}
+
+object SmoothDividedProgressBar extends SmoothProgressBar[SmoothDividedProgressBar] {
+  type Props = SmoothDividedProgressBar
+
+  implicit val propsReuse: Reusability[Props] = Reusability.derive[Props]
+
+  protected val component = ScalaComponent
+    .builder[Props]("SmoothDividedProgressBar")
+    .initialStateFromProps(State.fromProps)
+    .backend(x => new Backend(x))
+    .render_PS { (p, s) =>
+      DividedProgress(
+        sections = p.sections,
+        sectionTotal = p.sectionTotal,
+        value = s.value,
+        completeSectionColor = p.completeSectionColor,
+        ongoingSectionColor = p.ongoingSectionColor,
+        progressCls = p.progressCls,
+        barCls = p.barCls,
+        labelCls = p.labelCls
+        )
+    }
+    .componentDidMount(_.backend.setupTimer)
+    .componentWillReceiveProps($ =>
+                                 $.backend.newStateFromProps($.currentProps, $.nextProps))
+    .configure(TimerSupport.install)
+    .configure(Reusability.shouldComponentUpdate)
+    .build
+}
+
+final case class NodAndShuffleProgress(
+  summary: StepStateSummary,
+  sections: NodAndShuffleStatus => List[DividedProgress.Label],
+  quantitiesFromNodMillis: // (nsStatus, nodMillis) => (sectionTotal, currentValue)
+    (NodAndShuffleStatus, DividedProgress.Quantity) =>
+      (DividedProgress.Quantity, Option[DividedProgress.Quantity])
+) extends ReactProps {
+  @inline def render: VdomElement = NodAndShuffleProgress.component(this)
+
+  def isStopping: Boolean =
+    summary.tabOperations.stopRequested === StopOperation.StopInFlight
+
+  protected[steps] val connect =
+    SeqexecCircuit.connect(SeqexecCircuit.obsProgressReader(summary.obsId, summary.step.id))
+}
+
+object  NodAndShuffleProgress {
+  type Props = NodAndShuffleProgress
+
+  @Lenses
+  final case class State(progressConnect: ReactConnectProxy[Option[ObservationProgress]])
+
+  implicit val propsReuse: Reusability[Props] = Reusability.by(_.summary)

```

Comment:
This means we don't re-render on changes to `sections` and `quantities`. Is that because these don't change inside a step?
Eventually the user could change the definitions in the OT (e.g. change the amount of cycles) and resync. I believe this deep on the stack it would be refreshed anyway

Reply:
`sections` and `quantities` are functions, that's why I left them out of reusability.

Which actually means they are being created on each update, and we don't want that. I'm considering using reusable functions (https://github.com/japgolly/scalajs-react/blob/master/doc/PERFORMANCE.md#reusable-functions), but I'm afraid that in that case they may not get updated if the definitions change in the OT.

I'll look into the issue. I will probably try refactoring so that `NodAndShuffleProgress` is a superclass of `NodAndShuffleCycleProgress` and `NodAndShuffleNodProgress` instead of a child component.

Reply:
Ok, sounds good, I never could understand the reusable functions thing to be honest